evm/src/test/java/org/hyperledger/besu/evm/processor/AbstractMessageProcessorTest.java

```diff
@@ -161,10 +141,12 @@ void shouldTraceContextEnterExitForEip3155Test() {
             POST_EXECUTION, // STATICCALL
             CONTEXT_ENTER, // STATICCALL
             CONTEXT_EXIT, // STATICCALL
+            CONTEXT_ENTER, // Re-entry in root context
```

Comment:
This shouldn't happen.  The number of enters should match the number of exits.

Reply:
Perhaps move the trace inside the `NOT_STARTED` block?

Reply:
This is a re-entry in the caller context #0 after coming back from the callee context #1; I don't see what this is an issue?

Reply:
Or maybe we should have different hooks, _e.g._ `traceContextEnter` and `traceContextReEnter`, one when in the `NOT_STARTED` state, and the other one elsewhere?

Reply:
In this case, any context would have one enter, one exit, and as many re-entry as child frames.

Reply:
enter and re-enter have vastly different semantics.  enter should be idempotent.  re-enter can re-occur.